homeassistant/components/sensor/mysensors.py

```diff
@@ -0,0 +1,108 @@
+"""
+homeassistant.components.sensor.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MySensors sensors.
+
+Config:
+  sensor:
+    - platform: mysensors
+      port: '/dev/ttyACM0'
+"""
+import logging
+
+# pylint: disable=no-name-in-module, import-error
+import homeassistant.external.pymysensors.mysensors.mysensors as mysensors
+import homeassistant.external.pymysensors.mysensors.const as const
+from homeassistant.helpers.entity import Entity
+
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL, EVENT_HOMEASSISTANT_STOP, TEMP_CELCIUS)
+
+CONF_PORT = "port"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Setup the mysensors platform. """
+
+    devices = {}    # keep track of devices added to HA
+
+    def sensor_update(update_type, nid):
+        """ Callback for sensor updates from the MySensors gateway. """
+        sensor = gateway.sensors[nid]
+        if sensor.sketch_name is None:
+            return
+        if nid not in devices:
+            devices[nid] = MySensorsNode(sensor.sketch_name)
+            add_devices([devices[nid]])
+
+        node = devices[nid]
+        node.battery_level = sensor.battery_level
+        for child_id, child in sensor.children.items():
+            node.update_child(child_id, child)
+        node.update_ha_state()
+
+    port = config.get(CONF_PORT)
+    if port is None:
+        _LOGGER.error("Missing required key 'port'")
+        return False
+
+    gateway = mysensors.SerialGateway(port, sensor_update)
+    gateway.start()
+
+    # Just assume celcius means that the user wants metric for now.
+    # It may make more sense to make this a global config option in the future.
+    gateway.metric = (hass.config.temperature_unit == TEMP_CELCIUS)
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                         lambda event: gateway.stop())
+
+
+class MySensorsNode(Entity):
+    """ Represents a MySensors node. """
+    def __init__(self, name):
+        self._name = name
+        self.battery_level = 0
+        self.children = {}
+
+    @property
+    def should_poll(self):
+        """ MySensor gateway pushes its state to HA.  """
+        return False
+
+    @property
+    def name(self):
+        """ The name of this sensor. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return ''
```

Comment:
Ok, I'll try to refactor it to create a separate HA entity for each child sensor.  I was hoping to avoid cluttering up the UI by grouping related child sensors under their parent node, but it sounds like there are disadvantages to that approach.

In order to enable graphing functionality, the entity needs to implement unit_of_measurement, correct?  Are there any other requirements?

For an example to make sure I understand correctly, let's say that I have a light sensor -- it's unit_of_measurement should be '%' and the state is 0-100, right?